test-suite/src/test/scala/org/scalajs/testsuite/javalib/TreeSetTest.scala

```diff
@@ -0,0 +1,436 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib
+
+import java.util
+
+import scala.language.implicitConversions
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+
+import org.scalajs.jasminetest.JasmineTest
+
+import scala.scalajs.runtime.UndefinedBehaviorError
+
+import java.{util => ju}
+import ju.TreeSet
+
+object TreeSetTest extends JasmineTest {
```

Comment:
To get a better test coverage do this.
```scala
object TreeSetTest extends TreeSetTest(new TreeSetTestFactory)
class HashSetTest[F <: TreeSetTestFactory](treeSetTestFactory: F) extends AbstractSetTest[F](treeSetTestFactory) with SortedSetTest with NavigableSetTest {
  override def testApi(): Unit = {
    super.testApi()
    testSortedSetApi(treeSetTestFactory)
    testNavigableSetApi(treeSetTestFactory)
   
    it("should ... ") {
      // Tests specific to TreeSet
    }
    ...
  }
}
```
You will need to fill the tests in `SortedSetTest` and `NavigableSetTest` (this last one does not exist yet). You will have to move some of the tests from this class to one of those if it is testing that API.